cmake/modules/StandaloneCrt.cmake

```diff
@@ -40,129 +40,154 @@ if(MSVC)
-         "src/runtime/minrpc *.h -> src/runtime/minrpc"
-         "src/support generic_arena.h -> src/support"
-         "src/support ssize.h -> src/support"
-         "src/runtime/crt crt_config-template.h -> template"
-         )
-
-    set(standalone_crt_base "${CMAKE_CURRENT_BINARY_DIR}/standalone_crt")
-
-    foreach(job_spec IN LISTS CRT_FILE_COPY_JOBS)
-      string(REPLACE " " ";" job_spec "${job_spec}")
-      list(LENGTH job_spec job_spec_length)
-      math(EXPR job_spec_length_mod "${job_spec_length} % 3")
-      if(NOT "${job_spec_length_mod}" EQUAL 1)
-        message(FATAL_ERROR "CRT copy job spec list length is ${job_spec_length}; parsed job spec is ${job_spec}")
-      endif()
-      math(EXPR job_spec_stop "${job_spec_length} - 3")
-
-      list(GET job_spec 0 job_src_base)
-      set(job_src_base "${CMAKE_CURRENT_SOURCE_DIR}/${job_src_base}")
-      foreach(copy_pattern_index RANGE 1 "${job_spec_stop}" 3)
-        list(GET job_spec ${copy_pattern_index} copy_pattern)
-        math(EXPR copy_dest_index "${copy_pattern_index} + 2")
-        list(GET job_spec ${copy_dest_index} copy_dest)
-
-        tvm_file_glob(GLOB_RECURSE copy_files
-             RELATIVE "${job_src_base}"
-             "${job_src_base}/${copy_pattern}")
-        list(LENGTH copy_files copy_files_length)
-        if("${copy_files_length}" EQUAL 0)
-          message(FATAL_ERROR "CRT copy job matched 0 files: ${job_src_base}/${copy_pattern} -> ${copy_dest}")
-        endif()
-        foreach(copy_src IN LISTS copy_files)
-          get_filename_component(dest_path "${standalone_crt_base}/${copy_dest}/${copy_src}" ABSOLUTE)
-          tvm_micro_add_copy_file(host_isolated_build_deps ${job_src_base}/${copy_src} ${dest_path})
-        endforeach()
-      endforeach()
-    endforeach()
+    set(CRT_LIBRARY_NAME host_standalone_crt_${CRT_LIBRARY})
+    set(CRT_LIBRARY_SOURCES "")
 
-    add_custom_target(standalone_crt DEPENDS ${host_isolated_build_deps})
+    foreach(FILE_NAME IN LISTS ARGN)
+       list(APPEND CRT_LIBRARY_SOURCES ${FILE_NAME})
+    endforeach()
 
-    get_filename_component(host_build_dir_abspath "${CMAKE_CURRENT_BINARY_DIR}/host_standalone_crt" ABSOLUTE)
+    add_library(${CRT_LIBRARY_NAME}
+                STATIC
+                ${CRT_LIBRARY_SOURCES})
 
-    if(${VERBOSE})
-    set(make_quiet QUIET=)
-    else(${VERBOSE})
-    set(make_quiet )
-    endif(${VERBOSE})
+    # add this library to the list of CRT libraries
+    set(CRT_LIBRARIES ${CRT_LIBRARIES} ${CRT_LIBRARY_NAME} PARENT_SCOPE)
 
-    list(APPEND crt_libraries memory graph_executor microtvm_rpc_server microtvm_rpc_common common)  # NOTE: listed in link order.
-    foreach(crt_lib_name IN LISTS crt_libraries)
-      list(APPEND crt_library_paths "host_standalone_crt/lib${crt_lib_name}.a")
-    endforeach()
+    target_include_directories(${CRT_LIBRARY_NAME}
+                               PUBLIC
+                               ${CMAKE_CURRENT_SOURCE_DIR}/src/runtime/micro/
```

Comment:
update to use standalone_crt dir

Reply:
actually, this is only for crt_config.h so it's the same as what the previous Makefile did